apps-contrib/resources/openapi-generator.json

```
@@ -0,0 +1,9 @@
+{
+  "repositories": [
+    "central"
+  ],
+  "mainClass": "org.openapitools.codegen.OpenAPIGenerator",
```

This field can be removed, as it seems the manifest of `org.openapitools:openapi-generator-cli` specifies the main class already. This works in particular:
```text
$ cs launch org.openapitools:openapi-generator-cli:latest.stable
usage: openapi-generator-cli <command> [<args>]
...
```
```suggestion
```

Alternatively, if that's not the case for some older versions, a `?` suffix can be added, like
```suggestion
  "mainClass": "org.openapitools.codegen.OpenAPIGenerator?",
```
so that any main class from the manifest will take precedence over this one.

---

Thanks for the suggestion. I wrote code to prove that the Main-Class has not changed. And that has not changed).

```scala
import $ivy.`io.get-coursier::coursier:2.0.0-RC6-21`
import coursier._
import java.io.File
import java.io.FileInputStream
import scala.concurrent.ExecutionContext
def extractMainClassName(file: File): Option[String] = {
  val jarStream =
    new java.util.jar.JarInputStream(new FileInputStream(file))
  val mf = jarStream.getManifest
  Option(mf.getMainAttributes).flatMap(attrs =>
    Option(attrs.getValue("Main-Class"))
  )
}
def fetchDependencyMainClassUnsafe(d: Dependency): List[String] =
  Fetch()
    .addDependencies(d)
    .run()
    .filter(_.getName == s"${d.module.name.value}-${d.version}.jar")
    .toList
    .flatMap(extractMainClassName)
def fetchVersionsUnsafe(module: Module): List[String] =
  Versions()
    .withModule(module)
    .result()
    .unsafeRun()(ExecutionContext.global)
    .results
    .collect { case (_, Right(value)) => value }
    .flatMap(_.available)
    .toList
val module = Module(
  Organization("org.openapitools"),
  ModuleName("openapi-generator-cli")
)
fetchVersionsUnsafe(module)
  .foreach { version =>
    val mainClassOpt =
      fetchDependencyMainClassUnsafe(Dependency(module, version))
    println(s"$version: $mainClassOpt")
  }
```